src/devices_models/devices/renewable_generation.jl

```diff
@@ -4,231 +4,162 @@ struct RenewableFullDispatch <: AbstractRenewableDispatchFormulation end
+    )
+end
+
+function custom_reactive_power_constraints!(
+    psi_container::PSIContainer,
+    devices::IS.FlattenIteratorWrapper{T},
+    ::Type{RenewableConstantPowerFactor},
+) where {T <: PSY.RenewableGen}
     names = (PSY.get_name(d) for d in devices)
     time_steps = model_time_steps(psi_container)
-    p_var = get_variable(psi_container, ACTIVE_POWER, R)
-    q_var = get_variable(psi_container, REACTIVE_POWER, R)
+    p_var = get_variable(psi_container, ACTIVE_POWER, T)
+    q_var = get_variable(psi_container, REACTIVE_POWER, T)
     constraint_val = JuMPConstraintArray(undef, names, time_steps)
-    assign_constraint!(psi_container, REACTIVE_RANGE, R, constraint_val)
+    assign_constraint!(psi_container, REACTIVE_RANGE, T, constraint_val)
     for t in time_steps, d in devices
         name = PSY.get_name(d)
-        pf = sin(acos(PSY.get_powerfactor(d)))
+        pf = sin(acos(PSY.get_power_factor(d)))
         constraint_val[name, t] =
             JuMP.@constraint(psi_container.JuMPmodel, q_var[name, t] == p_var[name, t] * pf)
     end
     return
 end
 
-function activepower_constraints!(
-    psi_container::PSIContainer,
-    devices::IS.FlattenIteratorWrapper{R},
-    model::DeviceModel{R, <:AbstractRenewableDispatchFormulation},
-    system_formulation::Type{<:PM.AbstractPowerModel},
+function DeviceRangeConstraintSpec(
+    ::Type{<:RangeConstraint},
+    ::Type{ActivePowerVariable},
+    ::Type{T},
+    ::Type{<:AbstractRenewableDispatchFormulation},
+    ::Type{<:PM.AbstractPowerModel},
     feedforward::Union{Nothing, AbstractAffectFeedForward},
-) where {R <: PSY.RenewableGen}
-    parameters = model_has_parameters(psi_container)
-    use_forecast_data = model_uses_forecasts(psi_container)
-
-    if !parameters && !use_forecast_data
-        constraint_data = Vector{DeviceRange}(undef, length(devices))
-        for (ix, d) in enumerate(devices)
-            name = PSY.get_name(d)
-            ub = PSY.get_activepower(d)
-            limits = (min = 0.0, max = ub)
-            range_data = DeviceRange(name, limits)
-            add_device_services!(range_data, d, model)
-            constraint_data[ix] = range_data
-        end
-        device_range(
-            psi_container,
-            constraint_data,
-            constraint_name(ACTIVE_RANGE, R),
-            variable_name(ACTIVE_POWER, R),
+    use_parameters::Bool,
+    use_forecasts::Bool,
+) where {T <: PSY.RenewableGen}
+    if !use_parameters && !use_forecasts
+        return DeviceRangeConstraintSpec(;
+            range_constraint_spec = RangeConstraintSpec(;
+                constraint_name = make_constraint_name(
+                    RangeConstraint,
+                    ActivePowerVariable,
+                    T,
+                ),
+                variable_name = make_variable_name(ActivePowerVariable, T),
+                limits_func = x -> (min = 0.0, max = PSY.get_active_power(x)),
```

Comment:
should this be `get_max_active_power(x)`?

Reply:
No, this is the case for no_forecasts. It loads the current operating point in the system.